mypy/checkmember.py

```diff
@@ -474,6 +474,9 @@ def analyze_class_attribute_access(itype: Instance,
             msg.fail(messages.GENERIC_INSTANCE_VAR_CLASS_ACCESS, context)
         is_classmethod = ((is_decorated and cast(Decorator, node.node).func.is_class)
                           or (isinstance(node.node, FuncBase) and node.node.is_class))
+        if is_classmethod and not itype.type.tuple_type:
+            for base_itype in itype.type.bases:
+                t = expand_type_by_instance(t, base_itype)
```

Comment:
This is not what I would do bind all type variables. For example, it seems to me an extra subclassing will easily break this:
```python
class A(Generic[T]):
    @classmethod
    def m(cls) -> T: ...

class B(A[int]): ...
class C(B): ...
C.m()  # does this work? What about other cases of multiple inheritance?
```

Normally, (i.e. for binding type variables by instances) we use a combination `map_instance_to_supertype` and `expand_type_by_instance`. For this case I would write a similar more universal algorithm.